tests/ci/gtest_util.sh

```diff
@@ -22,7 +22,9 @@ function shard_gtest() {
 
     RESULT=0
     for PID in ${PIDS[*]}; do
-        if wait -f $PID; then
+        # The exit status of wait is the exit status of $PID
+        # `if` considers a zero exit status to be "true", but we need to branch on a non-zero exit status
+        if ! wait -f $PID; then
           RESULT=${?}
```

Comment:
Actually, this should be someone non-zero? Don't need the latest exist status assumption.